script/tracking.js

```diff
@@ -0,0 +1,179 @@
+var clicks = 0;
+var totalKeys = 0;
+var startMs;
+var totalHours = 0;
+var totalMinutes = 0;
+var totalSeconds = 0;
+
+var buttonClicks = 0;
+var userK = 0;
+var userC = 0;
+var emailK = 0;
+var emailC = 0;
+var pwK = 0;
+var pwC = 0;
+var fnameK = 0;
+var fnameC = 0;
+var lnameK = 0;
+var lnameC = 0;
+var addressK = 0;
+var addressC = 0;
+var countryK = 0;
+var countryC = 0;
+var zipcodeK = 0;
+var zipcodeC = 0;
+var sexK = 0;
+var sexC = 0;
+var languageK = 0;
+var languageC = 0;
+var aboutK = 0;
+var aboutC = 0;
+
+//track the mouse clicks
+document.onclick = function(){
+    clicks++;
+    document.getElementById("track-clicks").innerHTML="Number of mouse clicks: " + clicks;
+}
+
+//track total key presses
+document.onkeyup = function(){
+    totalKeys++;
+    document.getElementById("track-total-keys").innerHTML = "Total key presses: " + totalKeys;
+}
+
+//set the time of opening the page
+window.onload = function(){
+    startMs = Date.now();           //start time in milliseconds
+}
+
+function totalTime(){
+    var endMs = Date.now();
+
+    //calculate the total time in hours/minutes/seconds
+    var totalMs = endMs - startMs;
+    var fullSeconds = Math.floor(totalMs / 1000);
+
+    if(fullSeconds < 60){
+        totalSeconds = fullSeconds;
+    } else{
+        totalSeconds = fullSeconds % 60;
+        fullMinutes = Math.floor(fullSeconds / 60);
+
+        if(fullMinutes < 60){
+            totalMinutes = fullMinutes;
+        } else{
+            totalMinutes = fullMinutes % 60;
+            totalHours = Math.floor(fullMinutes / 60);
+        }
+    }
+
+    //display the time
+    document.getElementById("track-time").innerHTML = "Total time spent: " + totalHours + " hours, " + 
+        totalMinutes + " minutes and " + totalSeconds + " seconds";
+}
+
+//count total characters typed
+function totalCharacters(){
+    userC = document.getElementById("user").value.length;
+    emailC = document.getElementById("email").value.length;
+    pwC = document.getElementById("pw").value.length;
+    fnameC = document.getElementById("fname").value.length;
+    lnameC = document.getElementById("lname").value.length;
+    addressC = document.getElementById("address").value.length;
+    countryC = document.getElementById("country").value.length;
+    zipcodeC = document.getElementById("zipcode").value.length;
+    var about = document.getElementById("about").value;
+
+    //exclude the text in the optional box if nothing has been written by the user
+    if (about.includes("(Optional)Say something about yourself..")){
+        aboutC = 0;
+    } else{
+        aboutC = about.length;
+    }
+
+    var totalChar = userC + emailC + pwC + fnameC + lnameC + addressC + countryC + zipcodeC + aboutC;
+    document.getElementById("track-total-chars").innerHTML = "Total number of characters typed: " + totalChar;
+}
+
+//count the number of key presses per field
+function addKey(id){
+    document.getElementById(id).onkeyup = function(){
```

Comment:
for this part, I don't know if there is a better way, I will maybe ask TA on wednesday!